### PR TITLE
Use parse5 to serialise the DOM, use lodash to clone dom

### LINKED ISSUE #1101: find() convert `xlink:href` to `href`
`cheerio@^1.0.0-rc.2` automatically converts `xlink:href` to `href` when using `find()`.

```js
cheerio.load('')('<span><svg ><use xlink:href="#1" /></svg></span>').find('use').prop('xlink:href') //undefined
cheerio.load('')('<span><svg ><use xlink:href="#1" /></svg></span>').find('use').prop('href') //#1
```

Since, Safari doesn't support `href` without the "xlink" prefix in svg yet. Maybe, leave `xlink:href` as it was in `cheerio@^0.22.0` is better?

### PR DESCRIPTION
**Pull request by fb55**
Gives us a more capable serialiser.

This required the cloning logic to be replaced with lodash, as going through htmlparser2 will remove properties that parse5 relies on. That should also be considerably faster, so why not 😄

Fixes #1101

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -60,12 +58,21 @@ exports.domEach = function(cheerio, fn) {
...
+exports.cloneDom = function(dom) {
+  function filterOutParent(node) {
+    if (
+      "length" in dom
+        ? Array.prototype.some.call(dom, function(el) {
+          return node === el.parent;
+        })
+        : node === dom.parent
+    ) {
+      return null;
```
Comment (jugglinmike) [inline lib/utils.js]: This is tough to read syntactically, and the meaning is also a little hard to understand. What do you think about this:

```diff
+var parents = "length" in dom ?
+  Array.prototype.map.call(dom, function(el) { return el.parent; }) :
+  [dom.parent];
 function filterOutParent(node) {
-  if (
-    "length" in dom
-      ? Array.prototype.some.call(dom, function(el) {
-        return node === el.parent;
-      })
-      : node === dom.parent
-  ) {
+  if (parents.indexOf(node) > -1) {
     return null;
   }
 }
```

I have a feeling that `indexOf` will perform slightly better than `some`, but I'd recommend this even without that consideration.

File: lib/utils.js [hunk 2]
```diff
@@ -60,12 +58,21 @@ exports.domEach = function(cheerio, fn) {
...
+    }
+  }
 
-  return parse(render(dom, options), options, false).children;
+  return cloneDeepWith(dom, filterOutParent);
```
Comment (jugglinmike) [inline lib/utils.js]: The documentation still describes the heuristic you're replacing. Could you update that,too?

**Comment by jugglinmike** [comment]
This is cool!

**Comment by fb55** [comment]
Thanks for the review, updated!

**Comment by jugglinmike** [comment]
Merged to `v1.0.0` at commit e71d0ce6398c920a5c8d5b826f01f7a34da9d15a . Thanks, @fb55!

**Pull request closed without merging**